src/js/letters/components/Address.jsx

```diff
@@ -79,96 +48,113 @@ class Address extends React.Component {
           label="Country"
           name="country"
           autocomplete="country"
           options={this.props.countries}
           value={selectedCountry}
           required={this.props.required}
-          onValueChange={(update) => {this.props.onInput('country', update);}}/>
-        <ErrorableTextInput errorMessage={this.isValidAddressField(this.props.address.addressOne) ? undefined : 'Please enter a street address'}
+          onValueChange={(update) => this.props.onInput('country', update)}/>
+        <ErrorableTextInput errorMessage={errorMessages.addressOne}
           label="Street address"
           name="address"
           autocomplete="street-address"
           charMax={30}
           value={this.props.address.addressOne}
           required={this.props.required}
-          onValueChange={(update) => {this.props.onInput('addressOne', update);}}/>
+          onValueChange={(update) => this.props.onInput('addressOne', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
           charMax={30}
           value={this.props.address.addressTwo}
-          onValueChange={(update) => {this.props.onInput('addressTwo', update);}}/>
+          onValueChange={(update) => this.props.onInput('addressTwo', update)}/>
         <ErrorableTextInput
           label="Street address (optional)"
           name="address"
           autocomplete="street-address"
           charMax={30}
           value={this.props.address.addressThree}
-          onValueChange={(update) => {this.props.onInput('addressThree', update);}}/>
-        <ErrorableTextInput errorMessage={this.isValidAddressField(this.props.address.city) ? undefined : 'Please enter a city'}
+          onValueChange={(update) => this.props.onInput('addressThree', update)}/>
+        <ErrorableTextInput errorMessage={errorMessages.city}
           label={<span>City <em>(or APO/FPO/DPO)</em></span>}
           name="city"
           autocomplete="address-level2"
           charMax={30}
           value={this.props.address.city}
           required={this.props.required}
-          onValueChange={(update) => {this.props.onInput('city', update);}}/>
-        {stateProvince}
-        <ErrorableTextInput errorMessage={this.isValidPostalCode(this.props.address.zipCode) ? undefined : 'Please enter a valid Postal code'}
+          onValueChange={(update) => this.props.onInput('city', update)}/>
+        {/* Hide the state for addresses that aren't in the US */}
+        {isUSA && <ErrorableSelect errorMessage={errorMessages.state}
+          label="State"
+          name="state"
+          autocomplete="address-level1"
+          options={adjustedStateNames}
+          value={this.props.address.state}
+          required={this.props.required}
+          onValueChange={(update) => this.props.onInput('state', update)}/>}
+
+        {/* Hide the zip code for addresseses that aren't in the US */}
```

Comment:
I'm still not confident that we're handling this correctly. Even though `state` and `zipCode` aren't required for international addresses, I wonder if sending them will fail validation on the back-end. If not, and if they can be accepted, I think it better to still provide text fields for those questions so that someone can enter that information if they want to. Otherwise, it may not be a valid international address without it. I will add a ticket to check on this separately so it doesn't have to hold up this PR, but would like to return to this question.